src/Core/Usecase/Form/VerifyFormLoaded.php

```diff
@@ -58,26 +58,34 @@ protected function verifyFormExists()
 	 * Checks that the form exists.
 	 * @param  Data $input
 	 * @return void
+	 * @todo  maybe move to validator
 	 */
 	protected function verifyFormDoesNoExistInTargetedSurveyState()
 	{
 		// Ensure that the form exists.
 		if ($this->form_contact_repo->formExistsInPostStateRepo($this->getRequiredIdentifier('form_id'))) {
-			throw new \HTTP_Exception_400('The form already has a set of contacts');
```

Comment:
@rowasc FYI we should pretty much never be using HTTP errors directly in src/Core code.
The idea is that outer layers (controllers, commands etc) depend on inner layers (Usecases) but not the reverse.
So Controller -> Usecases -> Entities etc